src/main/java/com/google/javascript/gents/TypeAnnotationPass.java

```diff
@@ -147,37 +147,42 @@ public void visit(NodeTraversal t, Node n, Node parent) {
-            if (bestJSDocInfo != null) {
-              setTypeExpression(n, bestJSDocInfo.getType(), false);
-            }
+            maybeSetTypeExpression(n, bestJSDocInfo, false);
           } else if (parent.isParamList()) { // Function parameters
             JSDocInfo parentDocInfo = NodeUtil.getBestJSDocInfo(parent.getParent());
-            if (parentDocInfo == null) {
-              break;
-            }
-            JSTypeExpression parameterType = parentDocInfo.getParameterType(n.getString());
-            if (parameterType != null) {
-              Node attachTypeExpr = n;
-              // Modify the primary AST to represent a function parameter as a
-              // REST node, if the type indicates it is a rest parameter.
-              if (parameterType.getRoot().getToken() == Token.ELLIPSIS) {
-                attachTypeExpr = IR.rest(n.getString());
-                nodeComments.replaceWithComment(n, attachTypeExpr);
-                compiler.reportCodeChange();
-              }
-              // Modify the AST to represent an optional parameter
-              if (parameterType.getRoot().getToken() == Token.EQUALS) {
-                attachTypeExpr = IR.name(n.getString());
-                attachTypeExpr.putBooleanProp(Node.OPT_ES6_TYPED, true);
-                nodeComments.replaceWithComment(n, attachTypeExpr);
-                compiler.reportCodeChange();
+            if (parentDocInfo != null) {
+              JSTypeExpression parameterType = parentDocInfo.getParameterType(n.getString());
+              if (parameterType != null) {
+                // Parameter is declared using verbose @param syntax before the function definition.
+                Node attachTypeExpr = n;
+                // Modify the primary AST to represent a function parameter as a
+                // REST node, if the type indicates it is a rest parameter.
+                if (parameterType.getRoot().getToken() == Token.ELLIPSIS) {
+                  attachTypeExpr = IR.rest(n.getString());
+                  nodeComments.replaceWithComment(n, attachTypeExpr);
+                  compiler.reportCodeChange();
+                }
+                // Modify the AST to represent an optional parameter
+                if (parameterType.getRoot().getToken() == Token.EQUALS) {
+                  attachTypeExpr = IR.name(n.getString());
+                  attachTypeExpr.putBooleanProp(Node.OPT_ES6_TYPED, true);
+                  nodeComments.replaceWithComment(n, attachTypeExpr);
+                  compiler.reportCodeChange();
+                }
+                setTypeExpression(attachTypeExpr, parameterType, false);
+                break;
               }
-              setTypeExpression(attachTypeExpr, parameterType, false);
             }
+            // If we didn't break, then maybe the type is inlined just before the parameter?
+            maybeSetTypeExpression(n, bestJSDocInfo, false);
```

Comment:
It took me a bit to figure out what this does. Maybe it'd be more readable if we explicitly called `getType()` here, and check for `null`?